### PR TITLE
Add IPython display support

### LINKED ISSUE #304: Celiagg backend doesn't save images correctly
From enable, when I call `save` to put the contents of a celiagg gc into a .png file, I get this:

component_ex

where if I use the old agg backend (or indeed anything else), I get this:

component_ex

which is what I would expect based on the colors, etc. I was using.

Based on a little experimentation, the [channel shuffling](https://github.com/enthought/enable/blob/master/kiva/celiagg.py#L748) doesn't seem to be needed

### PR DESCRIPTION
**Pull request by corranwebster**
This PR adds two things:

- IPython/Jupyter Notebook support for the SVG, Agg and Celiagg backends (using `_repr_svg_` and `_repr_png_` respectively).
- a new `to_image` method for Agg and Celiagg backends which generates a PIL/Pillow Image object from the GC.

It also includes a drive-by fix for saving RGBA vs. BGRA format Celiagg graphics contexts, which fixes #304.

### CODE REVIEW

**Comment by jwiggins** [comment]
This will need to be rebased to account for the recent removal of `kiva.compat`. Also, #304 has been fixed by #592 (also a drive-by fix there...). And once #590 goes in, there might be some similar changes that can happen here.

Other than the image data in the example notebook, I have no big issues with this change.

**Comment by corranwebster** [comment]
This now has tests, so presuming test pass OK, this is probably ready for a proper review.

File: kiva/celiagg.py [hunk 1]
```diff
@@ -868,6 +856,52 @@ def save(self, filename, file_format=None, pil_options=None):
...
+            # Data is BGRA; Convert to RGBA
+            data = np.empty(pixels.shape, dtype=np.uint8)
+            data[..., 0] = pixels[..., 2]
+            data[..., 1] = pixels[..., 1]
+            data[..., 2] = pixels[..., 0]
+            data[..., 3] = pixels[..., 3]
+        else:
+            data = pixels
+
+        return Image.fromarray(data, 'RGBA')
```
Comment (jwiggins) [inline kiva/celiagg.py:886]: It hasn't been a problem yet, but we might want to add support for formats without alpha at some point.

File: kiva/examples/kiva/Kiva Explorer.ipynb [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
...LNOcMux5r7LptpaKRwghmiMJXZw1DYq2LwnXXUTyFBN6XxduxxFy1kQTPGAzhz/ax7Fl7R1jS5vN2NVplGxNo2Tr2T6hN7SF7K9aOi4hhGhIutzFWfOguneR9uVhMldeTI/f9SZuUk+iLwPIpeRXW+W9iAPP2nFVtXccQgghRKsIxi/hRkb86wGuWqb8Snt7gvGKae8YhBBCiDYRiG9se8fQFvRojGF4J4Tj09WA1tze8QghRH2/yqcqIVqSN3rLRBLv603oeB0aA4CK6kmjdNtq0t/LoVI2JBFCtDtJ6EKchhm9370MfMuCOaL2iKracVXp0Zq0aPQqqud7Ut/YTPbi9o1UCPFbJ0VxQpzGRBLutWCOyKBs1zqOfXyc8r1uPE4FlGC8YgYTde1ldH2wiOrjaZRua+94hRBCCNGExxix9FZ6P3e6or8xxN49hf4vt2VcQgjRkGzOIsRpGNCaV5A2TwW1uTabyP6yC7492jIuIYRoSBK6EKeRj/VoJY6i07XxQR/gwlPT8Hh/wi+vK6ITQojWJgldiNPYyPGF/Qm/vLnzZnS+k+j28HZyv214bjSxdyUSMLh1IxRCiFpSFCfEafQmbEIc/v0SCRzS8JwejbEL/r30aIw1eOy30vs5qN2cxQdDYADmyO4EjzxM8ca2j1wI8VsjCV2I07BgDDOh900meMTp2nUj6OKmjwcPB5mmLoRofZLQhTiNQxRtCMQc9Q1HXrJSU3q64jghhBBCdFCheMc/xLAF7R2HEEIIIS7Qr3XjGSGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEEIIIYQQQgghhBBCCCGEEG3i/wNa954mPWIbtQAAAABJRU5ErkJggg==\n",
```
Comment (jwiggins) [inline kiva/examples/kiva/Kiva Explorer.ipynb:30]: Can you leave this data out?

**Review by jwiggins — APPROVED** [review]
LGTM. Please strip the data out of the notebook before merging.

**Pull request merged**